### PR TITLE
[opt](inverted index) Optimize the compression of inverted index position information

### PR DESCRIPTION
**Pull request by zzzxl1993**
1. Optimize position information in the inverted index

### CODE REVIEW

**Comment by zzzxl1993** [comment]
run buildall

File: src/core/CLucene/util/PFORUtil.h [hunk 1]
```diff
@@ -18,9 +18,22 @@
...
+CL_NS_USE(store)
 
 size_t P4DEC(unsigned char *__restrict in, size_t n, uint32_t *__restrict out);
 size_t P4NZDEC(unsigned char *__restrict in, size_t n, uint32_t *__restrict out);
 size_t P4ENC(uint32_t *__restrict in, size_t n, unsigned char *__restrict out);
 size_t P4NZENC(uint32_t *__restrict in, size_t n, unsigned char *__restrict out);
 
+class PforUtil {
+public:
+    static constexpr size_t blockSize = 128;
```
Comment (airborne12) [inline src/core/CLucene/util/PFORUtil.h:35]: why change block size to 128 from 512?

File: src/core/CLucene/index/IndexWriter.cpp [hunk 2]
```diff
@@ -1308,20 +1309,36 @@ void IndexWriter::indexCompaction(std::vector<lucene::store::Directory *> &src_d
     }
     assert(readers.size() == numIndices);
 
-    // check hasProx
+    // check hasProx, indexVersion
     bool hasProx = false;
+    IndexVersion indexVersion = IndexVersion::kV1;
     {
         if (!readers.empty()) {
+            auto release_readers = [this]() {
```
Comment (airborne12) [inline src/core/CLucene/index/IndexWriter.cpp:1317]: why we release reader here?

File: src/core/CLucene/index/_SegmentHeader.h [hunk 3]
```diff
@@ -86,6 +87,62 @@ class TermDocsBuffer {
   IndexVersion indexVersion_ = IndexVersion::kV0; 
 };
 
+class TermPostingsBuffer {
```
Comment (airborne12) [inline src/core/CLucene/index/_SegmentHeader.h:90]: why we need this buffer? index input already has buffer inside.

**Comment by zzzxl1993** [comment]
run buildall

File: src/core/CLucene/index/IndexWriter.cpp (hunk 2 above)
Comment (zzzxl1993) [reply src/core/CLucene/index/IndexWriter.cpp]: Release memory in the try->catch->finally code

**Review by airborne12 — CHANGES_REQUESTED** [review]
Need clucene UT test here, for example, set IndexVersion v1 and v2

**Comment by zzzxl1993** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
lgtm

**Pull request merged**